src/content/1.7/basics/keeping_up-to-date/upgrade.md

```diff
@@ -0,0 +1,300 @@
+running them can be done with a specific PHP script.
+
+When you’re ready, run the file \`install/upgrade/upgrade.php\`.
+
+This can be done with a browser, by reaching the address
+`http://<shop_domain>/install/upgrade/upgrade.php`, or from your
+server's command line:
+
+```bash
+php install/upgrade/upgrade.php
+```
+
+In both cases, an XML log will be displayed. The result can be found in
+the attribute `result` of the first tag `<action>`:
+
+* `ok` if updates have been found and executed
+* `error` if something went wrong
+* `info` for next actions, displaying the details on the process
+
+#### Execution log
+
+When the upgrade script found some upgrades to apply, the SQL queries
+run will be listed along their respective result.
+
+```xml
+<?xml version="1.0" encoding="UTF-8"?><action result="ok" id="">
+<action result="info" id="1.7.0.5"><![CDATA[[OK] PHP 1.7.0.5 : /* PHP:ps_update_tabs(); */]]></action>
+<action result="info" id="1.7.0.5"><![CDATA[[OK] SQL 1.7.0.5 : ALTER TABLE `ps_currency` MODIFY `name` varchar(64) NOT NULL]]></action>
+<action result="info" id="1.7.1.0"><![CDATA[[OK] SQL 1.7.1.0 : SET SESSION sql_mode = '']]></action>
+<action result="info" id="1.7.1.0"><![CDATA[[OK] SQL 1.7.1.0 : SET NAMES 'utf8']]></action>
+[...]
+```
+
+You can double check that each action is marked as “OK”. If not,
+additional details will be shown after the request, which can help you
+fix the issue and re-execute the request manually on your database. In
+some cases, you may need to restore your database backup and start over.
+
+#### Error codes
+
+An error code can also be displayed. Each code is related to a message
+described here:
+
+* Error #27: The shop is running a newer version than the content provided by the install folder.
+* Error #28: The shop is already at the version you try to upgrade to.
+* Error #29: Could not find the current version. Check your database parameters file and the database connection.
+* Error #31: Unable to find upgrade directory in the installation path, does the folder `install/upgrade/sql` exist and is not empty?
+* Error #32: No upgrade needs to be applied.
+* Error #33: Error while loading a SQL upgrade file. Check your permissions of the folder `install/upgrade/sql`.
+* Error #40: The version provided in the file `install/install_version.php` is invalid.
+* Error #43: Error while upgrading database schema using doctrine.
+* Error #44: Error while updating translations.
+* Error #45: Error while enabling theme.
+
+### Cleanup
+
+Before going further, a few things should now be cleaned.
+
+-   The \`install\` folder, used to run the database upgrades, is not needed anymore and can be safely deleted.
+-   When opening your shop (in the front or back office) on your browser, you may see some visual issues. This can be due to your old assets being still served by a cache. Reload the them by force-refreshing the page(press ctrl+R on Windows / Linux or cmd+R on Mac OS) or clearing your browser’s cache.
+
```

Comment:
Some people say you need to regenerate the .htaccess file. Do you think it's necessary?

Reply:
The 1-click upgrade does it as well, indeed.

This can be added later.